### PR TITLE
feat: Add endpoints to sync GDC, GMC and Owner

### PR DESCRIPTION
**Pull request by Judge40**
The GMC details, GDC details and Person Owner should be synchronized
from their own endpoints.

TISNEW-5238

### CODE REVIEW

File: src/main/java/uk/nhs/hee/trainee/details/api/PersonOwnerResource.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+  /**
+   * Update the person owner for the trainee.
+   *
+   * @param tisId The ID of the trainee to update.
+   * @param dto   The person owner to update with.
+   * @return The updated PersonalDetails.
+   */
+  @PatchMapping("/{tisId}")
+  public ResponseEntity<PersonalDetailsDto> updateGmcDetails(
```
Comment (JosephKelly) [inline src/main/java/uk/nhs/hee/trainee/details/api/PersonOwnerResource.java:60]: Method name looks like it should be modified after C+P

**Review by JosephKelly — CHANGES_REQUESTED** [review]
I think there's just 1 small change.

I really like the refactoring of the `updatePersonalDetailsByTisId` to take the mapping function.

File: src/main/java/uk/nhs/hee/trainee/details/api/PersonOwnerResource.java (hunk 1 above)
Comment (Judge40) [reply src/main/java/uk/nhs/hee/trainee/details/api/PersonOwnerResource.java]: Done

**Review by JosephKelly — APPROVED** [review]
Gosh... I didn't realise we should be increasing versions even when it hasn't been merged in (as close as we get to a version 'release')

**Comment by Judge40** [comment]
> Gosh... I didn't realise we should be increasing versions even when it hasn't been merged in (as close as we get to a version 'release')

That'll be me not paying attention to not needing to bump it.

**Pull request merged**